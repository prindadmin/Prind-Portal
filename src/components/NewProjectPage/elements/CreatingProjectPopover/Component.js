import React, { Component } from 'react'
//import PropTypes from 'prop-types'

import PopOverHandler from '../../../common/popOverHandler'

import {
  Spinner,
  Intent,
} from '@blueprintjs/core'

import * as strings from '../../../../data/Strings'

export class Element extends Component {
  static propTypes = {
  }

  getContent = () => {
    return (
      <React.Fragment>
        {strings.PROJECT_CREATION_BILLING_WARNING}
        <Spinner size='150' intent={Intent.DANGER}/>
      </React.Fragment>
    )
  }

  render () {
    return (
      <PopOverHandler>
        <div id='popup-greyer' onClick={(e) => e.stopPropagation()}>
          <div id='project-creation-popover'>
            <div id='popup-box'>
              <div className='project-creation-popover-container' onClick={(e) => e.stopPropagation()}>
                <div className='element-title'>
                  {strings.PROJECT_CREATION_IN_PROGRESS}
                </div>
                <div className='element-description'>
                  {this.getContent()}
                </div>
              </div>
            </div>
          </div>
        </div>
      </PopOverHandler>
    )
  }

}

export default Element
