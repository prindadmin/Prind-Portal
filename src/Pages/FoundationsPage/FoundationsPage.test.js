import React from 'react';
import { shallow } from 'enzyme';
import ReactGA from 'react-ga';

import Component from './FoundationsPage'

beforeAll(() => {
  ReactGA.initialize('dummy', { testMode: true });
  global.open = jest.fn()
});

it('Foundations Page - Should render', () => {
  const props = {
    location: {
      pathname: "foundations"
    }
  };
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});

it('Foundations Page - Should render and click android buttons', () => {
  const props = {
    location: {
      pathname: "foundations"
    }
  };
  const component = shallow(<Component {...props} />);
  const androidButton = component.find('img[alt="Android Logo"]')
  androidButton.simulate('click')
  expect(global.open).toHaveBeenCalledWith("https://play.google.com/store/apps/details?id=com.buildingim.foundations", "_blank")
});

it('Foundations Page - Should render and click iOS buttons', () => {
  const props = {
    location: {
      pathname: "foundations"
    }
  };
  const component = shallow(<Component {...props} />);
  const iosButton = component.find('img[alt="iOS Logo"]')
  iosButton.simulate('click')
  expect(global.open).toHaveBeenCalledWith("https://apps.apple.com/us/app/foundations/id1499177355#?platform=iphone", "_blank")
});
