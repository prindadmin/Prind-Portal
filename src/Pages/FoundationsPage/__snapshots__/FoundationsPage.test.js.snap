// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Foundations Page - Should render 1`] = `
<div
  id="foundations-page"
>
  <div
    className="page-content-section row"
  >
    <div
      className="foundations-page-container full-height"
    >
      <div
        className="foundations-icon row"
      >
        <div
          className="col-12"
        >
          <img
            alt="Foundations Logo"
            src="images/Foundations-Logo.png"
            width="100%"
          />
        </div>
      </div>
      <div
        className="foundations-icon-text row"
      >
        <div
          className="col-12"
        >
          Foundations allows users to enter their data into a single system and control the access to it. By using Foundations as the single source of data, you keep complete control of your personally identifiable information while allowing those granted access to it to receive the most up-to-date information automatically while protecting against GDPR violations.
        </div>
      </div>
      <div
        className="foundations-download-icons row"
      >
        <div
          className="col-6"
        >
          <img
            alt="Android Logo"
            height="75"
            onClick={[Function]}
            src="images/Download_Android.svg"
          />
        </div>
        <div
          className="col-6"
        >
          <img
            alt="iOS Logo"
            height="75"
            onClick={[Function]}
            src="images/Download_iOS.svg"
          />
        </div>
      </div>
      <div
        className="foundations-download-text row"
      >
        <div
          className="col-6"
        />
        <div
          className="col-6"
        />
      </div>
    </div>
  </div>
</div>
`;
